tests/ra_aid/test_main.py

```
@@ -0,0 +1,123 @@
+from ra_aid.__main__ import parse_arguments
+from ra_aid.tools.memory import _global_memory
+from ra_aid.config import DEFAULT_RECURSION_LIMIT
+
+
+@pytest.fixture
+def mock_dependencies(monkeypatch):
+    """Mock all dependencies needed for main()."""
+    monkeypatch.setattr('ra_aid.__main__.check_dependencies', lambda: None)
+    
+    monkeypatch.setattr('ra_aid.__main__.validate_environment', 
+                        lambda args: (True, [], True, []))
```

Ruff (0.8.2)

Unused lambda argument: `args`

(ARG005)